### PR TITLE
provide ESM module, support tree-shaking

### PR DESCRIPTION
**Pull request by aarongranick-okta**
## Description:
- builds courage as a tree-shakeable ESM module
- builds ESM for distribution
- adds package exports
- does not break compatibility with existing commonJS configurations

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

- Did you verify the change by running downstream monolith artifacts? (YES | NO | UNSURE)

### Issue:

- [OKTA-369648](https://oktainc.atlassian.net/browse/OKTA-369648)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,15 +20,22 @@
     "not ie < 11",
     "not IE_Mob 11"
   ],
-  "main": "./target/js/okta-sign-in.js",
   "types": "./types/index.d.ts",
+  "main": "./target/js/okta-sign-in.js",
+  "module": "./target/esm/src/index.js",
+  "exports": {
+    "require": "./target/js/okta-sign-in.js",
+    "import": "./target/esm/src/index.js"
```
Comment (shuowu) [inline package.json:28]: nit: also add `"./package.json": "./package.json"`, so it can be accessed externally

**Review by shuowu — APPROVED** [review]

**Review by lesterchoi-okta — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (aarongranick-okta) [reply package.json:28]: Screen Shot 2022-04-25 at 4 27 39 PM

Seeing this error in editor, but not sure where it is coming from. Lint seems to be fine.

Comment (aarongranick-okta) [reply package.json]: Got this in downstream:

```
Error [ERR_INVALID_PACKAGE_CONFIG]: Invalid package config /root/okta/okta-auth-js/node_modules/@okta/okta-signin-widget/package.json. "exports" cannot contain some keys starting with '.' and some not. The exports object must either be an object of package subpath keys or an object of main entry condition name keys only.
```

Comment (shuowu) [reply package.json]: what about 
```
exports: {
  ".": {
    require: ...,
    import: ...
  },
  "./package.json": "./package.json"
}

**Pull request closed without merging**